lib/spack/spack/solver/asp.py

```diff
@@ -763,6 +759,30 @@ def key_fn(version):
                 )
             )
 
+        for v1 in most_to_least_preferred:
+            if isinstance(v1.version, spack.version.GitVersion):
+                # treat the reference as if if came from the package.py since that is the only
+                # acceptable place for a git matched version to have originated
+                ref_version = DeclaredVersion(
+                    version=spack.version.Version(v1.version.ref_version),
+                    idx=v1.idx,
+                    origin=version_provenance.package_py,
+                )
```

Comment:
This negates what we talked about with being able to extend it by putting the ref version in packages.yaml